src/components/AppController.jsx

```diff
@@ -3,30 +3,90 @@ import BaseComponent from "lib/BaseComponent";
 import { setAppReady } from "lib/falcor/falcorUtils";
 import ReactTransitionGroup from 'react-addons-transition-group';
 import _ from "lodash";
-import { TransitionManager } from "lib/loader";
+import { TransitionManager } from "lib/loader"
+import { setContextForGlobalErrorFunctions, resetContextForGlobalErrorFunctions, setGlobalError } from 'lib/utilities';
 
 // Components
 import Navigation from "components/Navigation";
 import Footer from "components/Footer";
-import Loader from "components/Loader"
+import Loader from "components/Loader";
+import Alert from 'components/Alert';
 
 // Application CSS; applicationStyles alias,
 // CSS and SCSS loaders in webpack.config.js
 require('applicationStyles');
 
 export default class AppController extends BaseComponent {
+  constructor(props) {
+    super(props);
+    this.safeSetState({
+      error: null,
+      displayErrorMessage: false,
+    });
+  }
+
   componentDidMount() {
+    setContextForGlobalErrorFunctions.call(this);
```

Comment:
This is a difficult function name, but in any case, I would move the context holder to Alert box